### ISSUE TITLE
src/test/regress/checkinc.py fails with XCode 10

### ISSUE THREAD

**Issue by jchampio** [greenplum-db/gpdb#7961]
(For reference, see [this issue comment](https://github.com/greenplum-db/gpdb/issues/7882#issuecomment-503367733), which I have spun this issue off from.)

`src/test/regress/checkinc.py` performs a check to ensure that the installed header tree is complete. However, once it gets to the system includes, it relies on their presence in `/usr/include`, and as of XCode 10, [those headers are no longer installed by default](https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes#3035624):

```
  ...
  utils/timestamp.h              includes missing file "math.h" 
  utils/timestamp.h              includes missing file "limits.h" 
  utils/varbit.h                 includes missing file "limits.h" 
  utils/zlib_wrapper.h           includes missing file "zlib.h" 
Include files... FAILED
```

Apple's documentation states
> If you are the maintainer of such software [requiring headers in `/usr/include`], we encourage you to update your project to work with the SDK or file a bug report for issues that are preventing you from doing so. As a workaround, an extra package is provided which will install the headers to the base system. In a future release, this package will no longer be provided.

[There have been other calls to remove this script in the past.](https://github.com/greenplum-db/gpdb/pull/2436)

**Comment by chrishajas** [comment]
As more people move to Mojave (which includes Xcode 10), they'll likely hit this issue. I've personally hit it and have seen others hit it too. GPDB still runs fine for me if I modify this script to do nothing, so its existence just requires another step for devs/contributors to get GPDB up and running.

**Comment by adam8157** [comment]
```
sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /
```

**Comment by danielgustafsson** [comment]
I have defended the script in the past since previous calls for removal mainly have focused on addressing a symptom and not a disease. That being said, the change that Apple has made seems to warrant removing it, as it is no longer clearcut what it should do (although I +1 the suggestion from  @adam8157 for any system regardless). At the end of the day, checking that system headers are in place is the job of autoconf and not the test harness.

**Comment by yydzero** [comment]
agree it is not test's job to check system headers. +1 to remove it

**Comment by adam8157** [comment]
Fixed by #8042
